pkg/xds/generator/outbound_proxy_generator.go

```diff
@@ -367,12 +367,23 @@ func (OutboundProxyGenerator) determineRoutes(
 			// We assume that all the targets are either ExternalServices or not
 			// therefore we check only the first one
 			var isExternalService bool
+			var hasEndpoints bool
 			if endpoints := proxy.Routing.OutboundTargets[service]; len(endpoints) > 0 {
+				hasEndpoints = true
 				isExternalService = endpoints[0].IsExternalService()
 			}
 			if endpoints := proxy.Routing.ExternalServiceOutboundTargets[service]; len(endpoints) > 0 {
+				hasEndpoints = true
 				isExternalService = true
 			}
+			if _, ok := proxy.Routing.ExternalServiceUnavailable[service]; ok {
```

Comment:
do we need this new field? can we just pass `xdsCtx.Mesh.ExternalServicesEndpointMap` to `determineRoutes` and do a lookup like here - https://github.com/kumahq/kuma/pull/8730/files#diff-f85d1997072108108f860d61dab0a92e0002f6edbdfab7be9faf0ef66ea89e42R196 ?

Reply:
Good point, yes, I think by using this existing data, we don't need to introduce a new field there.
I've just learnt more about the existing design. Thank you very much.

Reply:
I've just updated code. Thanks again.